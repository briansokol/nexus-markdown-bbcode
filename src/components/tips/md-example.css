.md-example {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    margin: 1rem 0;

    > div {
        margin-left: 1.5rem;
    }
}

.md-example-header {
    margin-top: 0.5rem;
    margin-bottom: 0.25rem;
}

.md-example-title {
    margin: 0.5rem 0 0;
    font-size: 1rem;
    font-weight: bold;
}

.md-example-input,
.md-example-output {
    margin: 0;
    font-size: 1rem;
}
