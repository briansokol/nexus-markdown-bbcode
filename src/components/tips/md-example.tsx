/** @jsxImportSource @emotion/react */
import * as styles from '@/components/tips/md-example.styles';
import { Markdown } from '@/markdown';

interface MdExampleProps {
    children: string | string[];
}

function addLineBreaks(children: string | string[]): string {
    if (Array.isArray(children)) {
        let newChildren = '';

        for (let i = 0; i < children.length; i++) {
            if (i > 0 && !children[i - 1].endsWith('\n') && !children[i].includes('\n')) {
                newChildren += '\n';
            }
            newChildren += children[i];
        }
        return newChildren;
    }
    return children;
}

export function MdExample({ children }: MdExampleProps) {
    console.log(addLineBreaks(children));
    return (
        <div css={styles.mdExample}>
            <h4 css={styles.mdExampleHeader}>Example:</h4>
            <div>
                <h5 css={styles.mdExampleTitle}>Example Markdown Input</h5>
                <pre css={styles.mdExampleInput}>{addLineBreaks(children)}</pre>
            </div>
            <div>
                <h5 css={styles.mdExampleTitle}>BBCode Output</h5>
                <pre css={styles.mdExampleOutput}>
                    <Markdown markdownInput={addLineBreaks(children)} mode="bbcode" />
                </pre>
            </div>
            <div>
                <h5 css={styles.mdExampleTitle}>HTML Output</h5>
                <Markdown markdownInput={addLineBreaks(children)} mode="html" />
            </div>
        </div>
    );
}
