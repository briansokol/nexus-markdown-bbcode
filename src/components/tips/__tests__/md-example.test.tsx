import { render } from '@/test/test-utils';
import { afterEach, describe, expect, it, vi } from 'vitest';
import { MdExample } from '../md-example';

// Mock console.log to avoid output in tests
const mockConsoleLog = vi.fn();
vi.spyOn(console, 'log').mockImplementation(mockConsoleLog);

describe('MdExample', () => {
    afterEach(() => {
        mockConsoleLog.mockClear();
    });

    /**
     * Test MdExample with simple string content
     */
    it('should render with simple string content', () => {
        const { container } = render(<MdExample>This is **bold** text</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with header markdown
     */
    it('should render with header markdown', () => {
        const { container } = render(<MdExample># Header Level 1</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with multiple paragraphs
     */
    it('should render with multiple paragraphs', () => {
        const { container } = render(
            <MdExample>This is a paragraph.{'\n\n'}This is another paragraph.</MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with color directive
     */
    it('should render with color directive', () => {
        const { container } = render(<MdExample>This is :red[red text]</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with alignment directive
     */
    it('should render with alignment directive', () => {
        const { container } = render(<MdExample>:center[Centered text]</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with list items
     */
    it('should render with unordered list', () => {
        const { container } = render(<MdExample>- Item 1</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with ordered list
     */
    it('should render with ordered list', () => {
        const { container } = render(<MdExample>1. Item 1</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with image markdown
     */
    it('should render with image markdown', () => {
        const { container } = render(
            <MdExample>![Official Addon](https://i.imgur.com/1FSEwwP.png)</MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with YouTube directive
     */
    it('should render with YouTube directive', () => {
        const { container } = render(<MdExample>{':youtube{v=tKWPfMLw3r8}'}</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with spoiler directive
     */
    it('should render with spoiler directive', () => {
        const { container } = render(
            <MdExample>
                :::spoiler{'\n'}This is a spoiler.{'\n\n'}Anything can go inside a spoiler.{'\n'}:::
            </MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with inline code
     */
    it('should render with inline code', () => {
        const { container } = render(<MdExample>`This is inline code`</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with code block
     */
    it('should render with code block', () => {
        const { container } = render(
            <MdExample>
                {`\`\`\``}
                This is a multi-line code block{'\n'}This is a multi-line code block
                {`\`\`\``}
            </MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with formatting styles
     */
    it('should render with various formatting styles', () => {
        const { container } = render(
            <MdExample>This is *italic* and ~~strikethrough~~ text</MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with link
     */
    it('should render with link', () => {
        const { container } = render(<MdExample>[Link Text](https://example.com)</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with underline directive
     */
    it('should render with underline directive', () => {
        const { container } = render(<MdExample>This is :ins[underline] text</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with caption directive
     */
    it('should render with caption directive', () => {
        const { container } = render(<MdExample>:small[This is a caption]</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with complex nested alignment
     */
    it('should render with complex nested alignment', () => {
        const { container } = render(
            <MdExample>
                :::center{'\n'}![Official Addon](https://i.imgur.com/1FSEwwP.png){'\n\n'}
                :small[Official Addon Banner]
                {'\n'}:::
            </MdExample>,
        );

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with array of strings
     */
    it('should render with array of strings', () => {
        const { container } = render(<MdExample>{['Line 1', 'Line 2', 'Line 3']}</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with mixed array content
     */
    it('should render with mixed array content including newlines', () => {
        const { container } = render(<MdExample>{['Line 1\n', 'Line 2', '\nLine 3']}</MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test MdExample with empty content
     */
    it('should render with empty content', () => {
        const { container } = render(<MdExample> </MdExample>);

        expect(container).toMatchSnapshot();
    });

    /**
     * Test that MdExample logs the processed content
     */
    it('should log the processed content to console', () => {
        render(<MdExample>Test content</MdExample>);

        expect(mockConsoleLog).toHaveBeenCalledWith('Test content');
    });

    /**
     * Test that MdExample logs array content correctly
     */
    it('should log array content with proper line breaks', () => {
        render(<MdExample>{['Line 1', 'Line 2']}</MdExample>);

        expect(mockConsoleLog).toHaveBeenCalledWith('Line 1\nLine 2');
    });
});
