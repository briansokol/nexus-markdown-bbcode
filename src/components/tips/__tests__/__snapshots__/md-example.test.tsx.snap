// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MdExample > should render with YouTube directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        :youtube{v=tKWPfMLw3r8}
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [youtube]
        tKWPfMLw3r8
        [/youtube]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          allowfullscreen=""
          height="315"
          src="https://www.youtube.com/embed/tKWPfMLw3r8"
          title="YouTube video player"
          width="560"
        />
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with alignment directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        :center[Centered text]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [
        center
        ]
        Centered text
        [/
        center
        ]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <span
          class="css-qjem27"
        >
          Centered text
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with array of strings 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        Line 1
Line 2
Line 3
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        Line 1 Line 2 Line 3
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        Line 1
Line 2
Line 3
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with caption directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        :small[This is a caption]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [size=
        2
        ]
        This is a caption
        [/size]
        

        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <span
          class="css-1oll1t"
        >
          This is a caption
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with code block 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        \`\`\`
This is a multi-line code block
This is a multi-line code block
\`\`\`
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [code]
        This is a multi-line code block
This is a multi-line code block

        [/code]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <pre
        class="pre-block"
      >
        <code
          class="css-xgh56q"
        >
          This is a multi-line code block
This is a multi-line code block

        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with color directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        This is :red[red text]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        This is 
        [color=
        #e06666
        ]
        red text
        [/color]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        This is 
        <span
          style="color: rgb(224, 102, 102);"
        >
          red text
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with complex nested alignment 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        :::center
![Official Addon](https://i.imgur.com/1FSEwwP.png)

:small[Official Addon Banner]
:::
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [
        center
        ]
        [size=
        3
        ]
        [img]
        https://i.imgur.com/1FSEwwP.png
        [/img]
        [/size]
        

        [size=
        3
        ]
        [size=
        2
        ]
        Official Addon Banner
        [/size]
        

        [/size]
        

        [/
        center
        ]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <span
        class="css-qjem27"
      >
        <p
          class="css-1bqejpo"
        >
          <img
            alt="Official Addon"
            class="css-v5z1y7"
            src="https://i.imgur.com/1FSEwwP.png"
          />
        </p>
        <p
          class="css-1bqejpo"
        >
          <span
            class="css-1oll1t"
          >
            Official Addon Banner
          </span>
        </p>
      </span>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with empty content 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
         
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      />
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with header markdown 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        # Header Level 1
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=5]
        [font=Trebuchet MS]
        [b]
        [color=#a5c4f3]
        HEADER LEVEL 1
        [/color]
        [/b]
        [/font]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <h1
        class="css-brkr6t"
      >
        Header Level 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with image markdown 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        ![Official Addon](https://i.imgur.com/1FSEwwP.png)
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [img]
        https://i.imgur.com/1FSEwwP.png
        [/img]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <img
          alt="Official Addon"
          class="css-v5z1y7"
          src="https://i.imgur.com/1FSEwwP.png"
        />
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with inline code 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        \`This is inline code\`
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [code]
        This is inline code
        [/code]
        

        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <code
          class="css-xgh56q"
        >
          This is inline code
        </code>
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with link 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        [Link Text](https://example.com)
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        [url=
        https://example.com
        ]
        Link Text
        [/url]
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        <a
          class="css-1pd9vk1"
          href="https://example.com"
        >
          Link Text
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with mixed array content including newlines 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        Line 1
Line 2
Line 3
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        Line 1 Line 2 Line 3
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        Line 1
Line 2
Line 3
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with multiple paragraphs 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        This is a paragraph.

This is another paragraph.
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        This is a paragraph.
        [/size]
        

        

        [size=
        3
        ]
        This is another paragraph.
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        This is a paragraph.
      </p>
      

      <p
        class="css-1bqejpo"
      >
        This is another paragraph.
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with ordered list 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        1. Item 1
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [list=1]
        

         
        [*][size=3]
        Item 1
        [/size]
        

         
        [/list]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <ol>
         
        <li>
          Item 1
        </li>
         
      </ol>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with simple string content 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        This is **bold** text
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        This is 
        [b]
        bold
        [/b]
         text
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        This is 
        <strong
          class="css-1vg6q84"
        >
          bold
        </strong>
         text
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with spoiler directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        :::spoiler
This is a spoiler.

Anything can go inside a spoiler.
:::
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [spoiler]
        [size=
        3
        ]
        This is a spoiler.
        [/size]
        

        [size=
        3
        ]
        Anything can go inside a spoiler.
        [/size]
        

        [/spoiler]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <div
        class="css-3gyvk1"
      >
        <p
          class="css-1bqejpo"
        >
          This is a spoiler.
        </p>
        <p
          class="css-1bqejpo"
        >
          Anything can go inside a spoiler.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with underline directive 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        This is :ins[underline] text
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        This is 
        [u]
        underline
        [/u]
         text
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        This is 
        <ins>
          underline
        </ins>
         text
      </p>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with unordered list 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        - Item 1
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [list]
        

         
        [*][size=3]
        Item 1
        [/size]
        

         
        [/list]
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <ul>
         
        <li>
          Item 1
        </li>
         
      </ul>
    </div>
  </div>
</div>
`;

exports[`MdExample > should render with various formatting styles 1`] = `
<div>
  <div
    class="css-h572dx"
  >
    <h4
      class="css-1j4b27w"
    >
      Example:
    </h4>
    <div>
      <h5
        class="css-17t88eq"
      >
        Example Markdown Input
      </h5>
      <pre
        class="css-q6c39y"
      >
        This is *italic* and ~~strikethrough~~ text
      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        BBCode Output
      </h5>
      <pre
        class="css-q6c39y"
      >
        [size=
        3
        ]
        This is 
        [i]
        italic
        [/i]
         and 
        [s]
        strikethrough
        [/s]
         text
        [/size]
        

      </pre>
    </div>
    <div>
      <h5
        class="css-17t88eq"
      >
        HTML Output
      </h5>
      <p
        class="css-1bqejpo"
      >
        This is 
        <em>
          italic
        </em>
         and 
        <del>
          strikethrough
        </del>
         text
      </p>
    </div>
  </div>
</div>
`;
